pkg/flow/internal/controller/loader.go

```diff
@@ -144,27 +119,26 @@ func (l *Loader) Apply(parentScope *vm.Scope, blocks []*ast.BlockStmt, configBlo
 			components = append(components, c)
 			componentIDs = append(componentIDs, c.ID())
 
-			if err = l.evaluate(logger, parentScope, c); err != nil {
+			if err = l.evaluateComponent(logger, parentScope, c); err != nil {
 				var evalDiags diag.Diagnostics
 				if errors.As(err, &evalDiags) {
 					diags = append(diags, evalDiags...)
 				} else {
 					diags.Add(diag.Diagnostic{
 						Severity: diag.SeverityLevelError,
 						Message:  fmt.Sprintf("Failed to build component: %s", err),
-						StartPos: ast.StartPos(n.(*ComponentNode).block).Position(),
-						EndPos:   ast.EndPos(n.(*ComponentNode).block).Position(),
+						StartPos: ast.StartPos(c.Block()).Position(),
+						EndPos:   ast.EndPos(c.Block()).Position(),
 					})
 				}
 			}
-		case *ConfigNode:
-			var errBlock *ast.BlockStmt
-			if errBlock, err = l.evaluateConfig(logger, parentScope, c); err != nil {
+		case BlockNode:
+			if err = l.evaluateConfigBlock(logger, parentScope, c); err != nil {
 				diags.Add(diag.Diagnostic{
 					Severity: diag.SeverityLevelError,
-					Message:  fmt.Sprintf("Failed to evaluate node for config blocks: %s", err),
-					StartPos: ast.StartPos(errBlock).Position(),
-					EndPos:   ast.EndPos(errBlock).Position(),
+					Message:  fmt.Sprintf("Failed to evaluate node for config block: %s", err),
+					StartPos: ast.StartPos(n.(BlockNode).Block()).Position(),
+					EndPos:   ast.EndPos(n.(BlockNode).Block()).Position(),
```

Comment:
Similar change here to the above:

```suggestion
					StartPos: ast.StartPos(c.Block()).Position(),
					EndPos:   ast.EndPos(c.Block()).Position(),
```